File: Open3D/08_icp/python/main.py
import open3d as o3d, numpy as np
s=o3d.io.read_point_cloud("source.ply"); t=o3d.io.read_point_cloud("target.ply"); threshold=0.05; result=o3d.pipelines.registration.registration_icp(s,t,threshold,np.eye(4),o3d.pipelines.registration.TransformationEstimationPointToPoint()); print(result); print(result.transformation)

File: Open3D/08_icp/cpp/main.cpp
#include <open3d/Open3D.h>
#include <iostream>
int main(){ auto s=open3d::io::CreatePointCloudFromFile("source.ply"), t=open3d::io::CreatePointCloudFromFile("target.ply"); if(!s||!t) return 1; auto r=open3d::pipelines::registration::RegistrationICP(*s,*t,0.05,Eigen::Matrix4d::Identity(),open3d::pipelines::registration::TransformationEstimationPointToPoint()); std::cout<<r.transformation_<<"\n"; }
